### PR TITLE
Add testcases for MPI_Exscan, MPI_Reduce_scatter and MPI_Scan

### PR DESCRIPTION
**Pull request by drwootton**
Add new tests to the existing set of collective-big-count tests.

A test for MPI_Alltoallv is being developed. This pull request should not be merged until this test is complete.

Signed-off-by: David Wootton <dwootton@us.ibm.com>

### CODE REVIEW

File: collective-big-count/test_exscan.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        ret += my_c_test_core(MPI_INT, V_SIZE_INT, false);
+        ret += my_c_test_core(MPI_C_DOUBLE_COMPLEX, V_SIZE_DOUBLE_COMPLEX, false);
+    }
+#else
+    size_t proposed_count;
+
+    // Each rank contribues: TEST_UNIFORM_COUNT elements
+    // Largest buffer is   : TEST_UNIFORM_COUNT elements
+    proposed_count = calc_uniform_count(sizeof(int), TEST_UNIFORM_COUNT,
+                                        2, 2, 1.0); // 1 send, 1 recv buffer each
```
Comment (jjhursey) [inline collective-big-count/test_exscan.c:40]: when you rebase you will need to remove the last parameter from the `calc_uniform_count` commands:
```suggestion
                                        2, 2); // 1 send, 1 recv buffer each
```

**Review by jjhursey — COMMENTED** [review]
This looks good. When you rebase this branch you will need to make a few minor edits to get it to build. Then I'll re-review.
I know you are working on the `alltoallv`. You can push that onto this branch and we can review them together if you like.

File: collective-big-count/test_exscan.c (hunk 1 above)
Comment (drwootton) [reply collective-big-count/test_exscan.c]: The MPI_Alltoallv test is complete, so implementation of this pull request is complete

**Review by jjhursey — APPROVED** [review]
Thanks for the new test cases!

**Pull request merged**